MultiSlider.js

```diff
@@ -1,534 +1,455 @@
+
+        const unconfined = I18nManager.isRTL ? this.state.pastOne - accumDistance : accumDistance + this.state.pastOne;
+        const bottom = 0;
+        const trueTop = this.state.positionTwo - (this.props.allowOverlap ? 0 : this.stepLength);
+        const top = trueTop === 0 ? 0 : trueTop || this.props.sliderLength;
+        const confined = unconfined < bottom ? bottom : unconfined > top ? top : unconfined;
+        const slipDisplacement = this.props.touchDimensions.slipDisplacement;
+
+        if (Math.abs(accumDistanceDisplacement) < slipDisplacement || !slipDisplacement) {
+            const value = positionToValue(confined, this.optionsArray, this.props.sliderLength);
+            const snapped = valueToPosition(value, this.optionsArray, this.props.sliderLength);
+            this.setState({
+                positionOne: this.props.snapped ? snapped : confined
+            });
+
+            if (value !== this.state.valueOne) {
+                this.setState(
+                    {
+                        valueOne: value
+                    },
+                    () => {
+                        let change = [this.state.valueOne];
+                        if (this.state.valueTwo) {
+                            change.push(this.state.valueTwo);
+                        }
+                        this.props.onValuesChange(change);
+                    }
+                );
+            }
+        }
+    };
+
+    moveTwo = gestureState => {
+        if (!this.props.enabledTwo) {
+            return;
+        }
+
+        const accumDistance = this.props.vertical ? -gestureState.dy : gestureState.dx;
+        const accumDistanceDisplacement = this.props.vertical ? gestureState.dx : gestureState.dy;
+
+        const unconfined = I18nManager.isRTL ? this.state.pastTwo - accumDistance : accumDistance + this.state.pastTwo;
+        const bottom = this.state.positionOne + (this.props.allowOverlap ? 0 : this.stepLength);
+        const top = this.props.sliderLength;
+        const confined = unconfined < bottom ? bottom : unconfined > top ? top : unconfined;
+        const slipDisplacement = this.props.touchDimensions.slipDisplacement;
+
+        if (Math.abs(accumDistanceDisplacement) < slipDisplacement || !slipDisplacement) {
+            const value = positionToValue(confined, this.optionsArray, this.props.sliderLength);
+            const snapped = valueToPosition(value, this.optionsArray, this.props.sliderLength);
+
+            this.setState({
+                positionTwo: this.props.snapped ? snapped : confined
+            });
+
+            if (value !== this.state.valueTwo) {
+                this.setState(
+                    {
+                        valueTwo: value
+                    },
+                    () => {
+                        this.props.onValuesChange([this.state.valueOne, this.state.valueTwo]);
+                    }
+                );
+            }
+        }
+    };
```

Comment:
Need some comments here about what's going on

Reply:
added for positionOne(pretty much the same here for positionTwo)